internal/provider/resource_vault_cluster.go

```diff
@@ -0,0 +1,153 @@
+package provider
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// defaultClusterTimeout is the amount of time that can elapse
+// before a cluster read operation should timeout.
+var defaultVaultClusterTimeout = time.Minute * 5
+
+// createUpdateTimeout is the amount of time that can elapse
+// before a cluster create or update operation should timeout.
+var createUpdateVaultClusterTimeout = time.Minute * 35
+
+// deleteTimeout is the amount of time that can elapse
+// before a cluster delete operation should timeout.
+var deleteVaultClusterTimeout = time.Minute * 25
+
+func resourceVaultCluster() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The Vault cluster resource allows you to manage an HCP Vault cluster.",
+		CreateContext: resourceVaultClusterCreate,
+		ReadContext:   resourceVaultClusterRead,
+		UpdateContext: resourceVaultClusterUpdate,
+		DeleteContext: resourceVaultClusterDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create:  &createUpdateVaultClusterTimeout,
+			Update:  &createUpdateVaultClusterTimeout,
+			Delete:  &deleteVaultClusterTimeout,
+			Default: &defaultVaultClusterTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"cluster_id": {
+				Description:      "The ID of the HCP Vault cluster.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"hvn_id": {
+				Description:      "The ID of the HVN this HCP Vault cluster is associated to.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// TODO: this is required for consul, but is this complete for vault yet? we can also mark computed output until it's ready.
```

Comment:
Need to confirm with Vault team.

Reply:
Sorry, what is required for consul? I'm unsure what this comment is referring to.

Reply:
Sorry, to rephrase: Is the `tier` feature complete for Vault clusters? Or is it still hidden behind a feature flag?
I ask because, if it's complete, I can include it as a required input on the cluster (like consul cluster does). If it's not complete, I can make it a computed output that gets automatically set to `development`. Or leave it out entirely until it's ready to be exposed.

Also, a more general question: what parts of the vault cluster API are currently feature flagged?

Reply:
TODO: based on some slack research, it looks like tiers can be used, so we'll make this a required input for now.

Reply:
Update: at this moment, `tier` can only be `development`, so making the a computed output for now.